Resources/skeleton/acme-demo-bundle/Acme/DemoBundle/Controller/SecuredController.php

```diff
@@ -20,15 +21,24 @@ class SecuredController extends Controller
      */
     public function loginAction(Request $request)
     {
-        if ($request->attributes->has(SecurityContext::AUTHENTICATION_ERROR)) {
-            $error = $request->attributes->get(SecurityContext::AUTHENTICATION_ERROR);
+        if (class_exists('\Symfony\Component\Security\Core\Security')) {
+            $authErrorKey = Security::AUTHENTICATION_ERROR;
+            $lastUsernameKey = Security::LAST_USERNAME;
         } else {
-            $error = $request->getSession()->get(SecurityContext::AUTHENTICATION_ERROR);
+            // BC for SF < 2.6
+            $authErrorKey = SecurityContext::AUTHENTICATION_ERROR;
+            $lastUsernameKey = SecurityContext::LAST_USERNAME;
+        }
+
+        if ($request->attributes->has($authErrorKey)) {
+            $error = $request->attributes->get($authErrorKey);
+        } else {
+            $error = $request->getSession()->get($authErrorKey);
         }
 
         return array(
-            'last_username' => $request->getSession()->get(SecurityContext::LAST_USERNAME),
-            'error'         => $error,
+            'last_username' => $request->getSession()->get($lastUsernameKey),
+            'error' => $error,
         );
```

Comment:
You can simplify that even more as of 2.7. See http://symfony.com/doc/current/cookbook/security/form_login_setup.html